app/src/main/java/com/github/sdpteam15/polyevents/fragments/MapsFragment.kt

```diff
@@ -34,9 +44,367 @@ class MapsFragment : Fragment() {
+            addArea("2", listEvent2, "Triangle")
+
+            val listEvent3 = arrayListOf<LatLng>()
+            listEvent3.add(LatLng(46.52111073013754, 6.565624214708805))
+            listEvent3.add(LatLng(46.52107750943789, 6.565624214708805))
+            listEvent3.add(LatLng(46.52108443041863, 6.566078178584576))
+            listEvent3.add(LatLng(46.521115113422766, 6.5660761669278145))
+            listEvent3.add(LatLng(46.521115113422766, 6.565871313214302))
+            listEvent3.add(LatLng(46.52115986905187, 6.565871313214302))
+            listEvent3.add(LatLng(46.52115986905187, 6.565824374556541))
+            listEvent3.add(LatLng(46.521115113422766, 6.565824374556541))
+            addArea("3", listEvent3, "La route en T")
+        }
+    }
+
+    /**
+     * Changes the style of the map
+     */
+    private fun setMapStyle() {
+        map!!.setMapStyle(MapStyleOptions(resources.getString(R.string.style_test3)))
+    }
+
+    /**
+     * Helper method to add a area to the map and generate an invisible marker in its center to display the area infos
+     */
+    private fun addArea(id: String, coords: List<LatLng>, name: String) {
+        if (!coords.isEmpty()) {
+            val poly = PolygonOptions()
+            poly.addAll(coords).clickable(true)
+
+            val polygon = map!!.addPolygon(poly)
+            polygon.tag = id
+
+            var list = coords
+            var lat = 0.0
+            var lng = 0.0
+            if (list.first() == list.last()) {
+                list = coords.subList(0, coords.size - 1)
+            }
+            for (coord in list) {
+                lat += coord.latitude
+                lng += coord.longitude
+            }
+
+            val center = LatLng(lat / list.size, lng / list.size)
+
+            val marker: Marker = map!!.addMarker(
+                MarkerOptions()
+                    .position(center)
+                    .title(name)
+                    .icon(getMarkerIcon())
+            )
+            areasPoints.put(id, Pair(marker, polygon))
+        }
+    }
+
+    /**
+     * Generates the icon for the invisible icons
+     * ref : https://stackoverflow.com/questions/35718103/how-to-specify-the-size-of-the-icon-on-the-marker-in-google-maps-v2-android
+     * TODO : Check if we should make it a singleton to save memory/performance
+     */
+    private fun getMarkerIcon(): BitmapDescriptor {
+        val vectorDrawable: Drawable? = ContextCompat.getDrawable(context!!, R.drawable.ic_location)
+        vectorDrawable?.setBounds(0, 0, 0, 0)
+        val bitmap = Bitmap.createBitmap(1, 1, Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(bitmap)
+        vectorDrawable?.draw(canvas)
+        return BitmapDescriptorFactory.fromBitmap(bitmap)
+    }
+
+    /**
+     * Sets the minimal and the maximal zoom
+     */
+    private fun setMinAndMaxZoom() {
+        map!!.setMinZoomPreference(18f)
+        map!!.setMaxZoomPreference(20f)
```

Comment:
Maybe we could make a list of constant at the top of the application with the fix value so that it is easier to change